zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -54,208 +47,148 @@ pub struct SignVerifyChip<F: Field, const MAX_VERIF: usize> {
+            let is_address_zero = meta.query_advice(main_gate_config.advices()[0], Rotation::cur());
+            let is_enable = q_keccak * not::expr(is_address_zero);
+
+            let input = [
+                is_enable.clone(),
+                is_enable.clone() * meta.query_advice(rlc, Rotation::cur()),
+                is_enable.clone() * 64usize.expr(),
+                is_enable * meta.query_advice(rlc, Rotation::next()),
+            ];
+            let table = [
+                keccak_table.is_enabled,
+                keccak_table.input_rlc,
+                keccak_table.input_len,
+                keccak_table.output_rlc,
+            ]
+            .map(|column| meta.query_advice(column, Rotation::cur()));
+
+            input.into_iter().zip(table).collect()
+        });
+
         Self {
-            q_enable,
-            pk_hash,
-            address,
-            msg_hash_rlc,
-            address_is_zero,
-            address_inv,
             range_config,
             main_gate_config,
-            pk,
-            msg_hash,
-            power_of_randomness,
             keccak_table,
+            q_rlc_evm_word,
+            q_rlc_keccak_input,
+            rlc,
+            q_keccak,
         }
     }
-}
 
-pub(crate) struct KeccakAux {
-    input: [u8; 64],
-    output: [u8; 32],
+    #[rustfmt::skip]
+    fn configure_rlc<F: Field>(
+        meta: &mut ConstraintSystem<F>,
+        name: &'static str,
+        main_gate_config: MainGateConfig,
+        q_rlc: Selector,
+        rlc: Column<Advice>,
+        challenge: Expression<F>,
+    ) {
+        // Layout (take input with length 12 as an example)
+        // | q_rlc |                          rlc                        |   a   |   b   |   c   |   d    |   e    |
+        // | ----- | --------------------------------------------------- | ----- | ----- | ----- | ------ | ------ |
+        // |   1   |                                                   0 |     0 |     0 |     0 |  be[0] |  be[1] |
+        // |   1   |                                  be[0]*r^1 +  be[1] | be[2] | be[3] | be[4] |  be[5] |  be[6] |
+        // |   1   | be[0]*r^6  + be[1]*r^5  + ... +  be[5]*r^1 +  be[6] | be[7] | be[8] | be[9] | be[10] | be[11] |
+        // |   1   | be[0]*r^11 + be[1]*r^10 + ... + be[10]*r^1 + be[11] |       |       |       |        |        |
+        meta.create_gate(name, |meta| {
+            let q_rlc = meta.query_selector(q_rlc);
+            let [a, b, c, d, e] = main_gate_config
+                .advices()
+                .map(|column| meta.query_advice(column, Rotation::cur()));
+            let [rlc, rlc_next] = [Rotation::cur(), Rotation::next()]
+                .map(|rotation| meta.query_advice(rlc, rotation));
+            let inputs = [e, d, c, b, a, rlc];
+            let powers_of_challenge = iter::successors(challenge.clone().into(), |power| {
+                (challenge.clone() * power.clone()).into()
```

Comment:
In this expression we end up with `r^5`.  Does this affect the maximum expression degrees in the constraints?  Or since it comes from a challenge expression it has no extra cost that could affect the extended degree?

Reply:
> since it comes from a challenge expression it has no extra cost that could affect the extended degree?

Yes, it's just like a constant.